src/tests/functions/New-PuppetDscModule.Tests.ps1

```diff
@@ -0,0 +1,275 @@
+Describe "New-PuppetDscModule" {
+  InModuleScope puppet.dsc {
+    Context "Basic functionality" {
+      Mock Get-PuppetizedModuleName {$Name.ToLowerInvariant()}
+      Mock Initialize-PuppetModule {}
```

Comment:
I think Pester 5 might give us a way to mock at a higher level that will fall through child Contexts so you don't have to repeat this for every damn context block. That would be nice.

Reply:
This is one of those times I really wish pester was more like rspec, I would like it to inherit mock definitions but reset how many times it's been called, or track calls in this context separately or something.